app/src/tests/specmath.service.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { Observable } from 'rxjs';
+import { SpecMathService } from '../shared/services/specmath.service';
+import { processFilesMockRequest } from './mocks/mockRequests';
+import { processFilesMockResponseNonConflict, processFilesMockResponseConflict, processFilesMockResponseError } from './mocks/mockResponses';
+import { routes } from '../shared/routes';
+
+describe('SpecMathService', () => {
+  let httpMockObject: HttpTestingController;
+  let service: SpecMathService;
+  let mockProcessFilesCall: Observable<any>;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [SpecMathService],
+    });
+    httpMockObject = TestBed.get(HttpTestingController);
+    service = TestBed.get(SpecMathService);
+
+    mockProcessFilesCall = service.uploadFiles(
+      processFilesMockRequest.specs,
+      processFilesMockRequest.operation,
+      processFilesMockRequest.driverFile
+    );
+  });
+
+  it('should be instantiated', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('uploadFiles() should return an object with a result spec when non-conflicting specs are sent', () => {
+    mockProcessFilesCall.subscribe((res: object) => {
+      expect(res).toEqual(processFilesMockResponseNonConflict);
+    });
+
+    const mockRequest = httpMockObject.expectOne(routes.processFiles);
+    expect(mockRequest.request.method).toBe('POST');
+    mockRequest.flush(processFilesMockResponseNonConflict);
+  });
+
+  it('uploadFiles() should return an object with a list of merge conflicts when conflicting specs are sent', () => {
```

Comment:
I don't see these additional tests adding any value - they all test one behavior, that the service returns the result from the http client.

Reply:
Addressed in commit `d972f64`